app.py

```diff
@@ -1,18 +1,27 @@
 import requests
-from flask import Flask, session, render_template, redirect, url_for
+from flask import Flask, session, render_template, redirect, url_for, request
 import os
 
 app = Flask(__name__)
 
+
 app.secret_key = os.environ['secret_key']
 
 
 @app.route('/')
 def location():
     if (session and session['ip_info']):
-        data = session['ip_info']
```

Comment:
We still should store ip information in the session (once it has been retrieved by this new method). Otherwise there's no need to keep a seperate condition for having `session['ip_info']` and not having `session['ip_info']` if we aren't using that stored information to avoid an additional API call.